oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,210 @@
+  once the translations are reviewed and accepted.
+* Transifex only allows a one-to-one relationship between repositories and Transifex
+  Projects. Organizing all of the translation files into one repository and one Transifex
+  Project has a lower cost: we pay by the project so we end up paying less, and by
+  putting all translation files in the same place, it is easier to track translation
+  progress, and debug translation issues.
+* A repository that only contains text/binary files, and uses branches to separate
+  translations related to Open edX releases can make all interactions with translations
+  very quick and simple due to the ability to clone the branch of a specific release,
+  with translations organized by repository name.
+* By using an app that is maintained by Transifex the organization, we reduce the
+  maintenance burden and are more future proof of changes they might make since they
+  maintain both the API and the github App.
+
+Proposed Implementation
+***********************
+
+Move translation files to a new repo
+====================================
+
+Translation files (of types .mo and .po) currently exist amongst the code/documentation
+they translate. We will move these Translation files from being amongst the
+code/documentation to their own repo. For easier reintegration, Translation files will be
+kept in the same directory structure as the code/documentation they translate.
+
+Add GitHub Transifex App to openedx organization
+================================================
+
+Transifex's GitHub app will need to be added to the openedx GitHub organization in order
+to grant the app permissions to push/pull the Translation files. Currently, we manage the
+push/pull permissions for the edx-transifex-bot through a number of groups. The Transifex
+GitHub app is granted permission on a repository basis, and by moving all the
+Translations to a single repository we reduce the number of user groups we will have to
+manage.
+
+Connect i18n repository to Transifex via app
+============================================
+
+The Transifex webapp accepts configuration files for each Transifex project. By
+connecting the single repository containing all Translation files, we only need to make a
+single configuration file that allows the Transifex GitHub app to manage the Translation
+files. Based on the Translation Working Group's instruction, we can set parameters that
+automatically push and pull Translation files.
+
+Copy Transifex Memory
+=====================
+
+As a last step we can save all the progress the Open edX translators have accomplished by
+copying the Transifex Memory, the auto-translation feature that allows for Projects with
+similar strings to be automatically translated, from the old projects to this new one. By
+moving all the Translation Files to the same repository we can increase the reach of the
+Transifex Memory feature to help translate similar strings across the entire
+code/documentation base.
```

Comment:
wait, are we making a new transifex project, and if so, will existing translators/reviewers maintain their access?

Reply:
We are making a new transifex project to track the new repo. Existing translations/reviewers will be granted access to this new project given their involvement in a Transifex project that is being made redundant by this OEP.

This has been added to this paragraph.